clang/test/CodeGenSYCL/device_has.cpp

```diff
@@ -45,13 +45,21 @@ void foo() {
   q.submit([&](handler &h) {
     KernelFunctor f1;
     h.single_task<class kernel_name_1>(f1);
-    // CHECK: define dso_local spir_kernel void @{{.*}}kernel_name_2{{.*}} !sycl_declared_aspects ![[ASPECTS4:[0-9]+]]
+    // CHECK: define dso_local spir_kernel void @{{.*}}kernel_name_2{{.*}} !sycl_declared_aspects ![[ASPECTS4:[0-9]+]] !srcloc ![[SRCLOC8:[0-9]+]]
     h.single_task<class kernel_name_2>([]() [[sycl::device_has(sycl::aspect::gpu)]] {});
   });
 }
 
 // CHECK: [[ASPECTS1]] = !{i32 1}
+// CHECK: [[SRCLOC1]] = !{i32 {{[0-9]+}}}
```

Comment:
Maybe it makes sense to check that `srcloc` metadata contains correct source line?

Reply:
The source locations are encoded, so though we could just use the encoding, it would be difficult to adjust if the tests are ever changed.

Reply:
Ok, fair enough.